docs/PROXY_WASM.md

```diff
@@ -749,6 +749,11 @@ payloads.
    eviction mechanism when the allocated memory slab is full:
     - `proxy_enqueue_shared_queue`
 
+5. In ngx_wasm_module, metric names are prefixed with `pw.filter_name.`, where
+   `filter_name` represents the name assigned to the filter in which the metric
+   is defined.
+   In Envoy, however, metric names are prefixed with `wasmcustom.`.
```

Comment:
I think it may be difficult for users to understand when specifically this difference matters. Am I right that it only matters to them when retrieving metrics by name through the FFI `get_metric_by_name`? Since in the SDK metrics are retrieved by id only?

Reply:
That's right.

I included this paragraph after inspecting Envoy's response on its admin metrics endpoint `/metrics` and seeing that it was exposing metrics with their internal prefixing, e.g. `wasmcustom.waf_filter.tx.total`, instead of just `waf_filter.tx.total`. 

But I hadn't noticed the `/metrics/prometheus` admin endpoint which properly exposes the metric names without their internal prefixing, e.g. `waf_filter_tx_total`.

Yeah, I think the user can be spared the details of internal prefixing if they never have to deal with them.

Since this information is only relevant to the FFI `get_metric_by_name` users , perhaps we should remove the paragraph entirely from `PROXY_WASM.md` and leave it as a comment in the FFI code.

Reply:
Hmmm yes, I think we should document this in he FFI. As for removing it from our public PROXY_WASM.md documentation, I agree, but I also think we should keep track of all our internal differences with other host SDK implementations. Maybe a `.md` document in `assets/`?

Reply:
I think I will do this locally to speed up the merge - if you haven't started yet.

Reply:
Yeah, I have. I messaged you privately to confirm it would be okay to:
- drop the doc commit entirely
- amend the FFI commit with the comment about prefixing
- add a separate commit containing a new `assets/INTERNAL_DIFFERENCES.md` file.

The changes are ready.

Reply:
Ok let's do it then!